src/Microsoft.AspNet.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -158,65 +192,85 @@ public override TagHelperContent AppendFormat([NotNull] string format, params ob
             [NotNull] string format,
             params object[] args)
         {
-            _buffer.Add(string.Format(provider, format, args));
+            Buffer.Append(string.Format(provider, format, args));
             return this;
         }
 
         /// <inheritdoc />
-        public override TagHelperContent Append(TagHelperContent tagHelperContent)
+        public override TagHelperContent Append(IHtmlContent htmlContent)
         {
-            if (tagHelperContent != null)
-            {
-                foreach (var value in tagHelperContent)
-                {
-                    Append(value);
-                }
-            }
-
-            // If Append() was called with an empty TagHelperContent IsModified should
-            // still be true. If the content was not already modified, it means it is empty.
-            // So the Clear() method can be used to indirectly set the IsModified.
-            if (!IsModified)
-            {
-                Clear();
-            }
-
+            Buffer.Append(htmlContent);
             return this;
         }
 
         /// <inheritdoc />
         public override TagHelperContent Clear()
         {
-            _buffer.Clear();
+            Buffer.Clear();
             return this;
         }
 
         /// <inheritdoc />
         public override string GetContent()
```

Comment:
This method is likely going to be used less than it was. But it'll be very useful when user code needs to check existing content e.g. where the `<option/>` tag helper compares attribute values against known values.

To be truly useful however, should allow caller to pass an `IHtmlEncoder` instance e.g. something analogous to the `NullTestEncoder`. This would ease comparisons that HTML encoding currently messes up.

But I think this is a follow-up issue. If you agree, please file the issue.